src/fileio.jl

```diff
@@ -24,7 +24,7 @@ function savefile(file::TextFile, content)
     write(file.filename, string(content))
 end
 
-function query_extended(filename)
+function query_extended(filename::AbstractString)
```

Comment:
Why?

Reply:
This function expects only string types `splitext`, I added this to make sure nothing strange is passed here as a safety net.